script.js

```diff
@@ -1,12 +1,442 @@
+const showSearchInputHtml = document.getElementById("show-search-input");
+const episodeSearchResultDisplayHtml = document.querySelector(".display-episode-search");
+const showSearchResultDisplayHtml = document.querySelector(".display-show-search");
+const episodeSelectHtml = document.querySelector("#episode-select-html");
+const showSelectHtml = document.querySelector("#show-select-html");
+const showListingButton = document.querySelector("#show-listing-button");
+
+// Global variables
+let allEpisodes;
+let allShows;
+
+window.onload = setup;
+
+// Fetch episodes function
 function setup() {
-  const allEpisodes = getAllEpisodes();
-  makePageForEpisodes(allEpisodes);
+  allShows = getAllShows().sort((a, b) => {
+    return a.name.localeCompare(b.name); // sort shows into alphabeltical order by comparing string, not numbers
+  });
+
+  buildShowDropdownList(allShows);
+  makePageForShows(allShows);
+}
+
+function fetchShowEpisodes(showId) {
+  fetch(`https://api.tvmaze.com/shows/${showId}/episodes`) // returns a promise and it is pending
+    .then((response) => {
+      if (response.status >= 200 && response.status <= 299) {
+        return response.json(); // returns the promise as fulfilled and gives us a response object which we pass into the callback function
+      } else {
+        throw new Error(
+          `Encountered something unexpected: ${response.status} ${response.statusText}`
+        );
+      }
+    })
+    .then((data) => {
+      allEpisodes = data;
+      showSearchInputHtml.value = "";
+      makePageForEpisodes(allEpisodes);
+      buildEpisodeDropdownList(allEpisodes);
+    })
+    .catch((error) => {
+      console.log(error);
+    });
+}
+
+// level 100
+
+/*
+  If the user clicks show select then the show select will disappear
+  If the user clicks the see show listing button then the episodes will disappear and the shows will appear also the episode list should not be shown initially
+*/
+
+function toggleVisibility(showElement, hideElement) {
+  showElement.style.display = "inline"; // makes the element visible
+  hideElement.style.display = "none"; // hides the element
+}
+
+function getFormattedEpisodeNameInDropdown(episode) {
+  const episodeTitle = episode.name;
+  const paddedSeasonNum = ("0" + episode.season).slice(-2);
+  const paddedEpisodeNum = ("0" + episode.number).slice(-2);
+  return `S${paddedSeasonNum}E${paddedEpisodeNum} - ${episodeTitle}`;
+}
+
+function getFormattedEpisodeNameInEpisodeCards(episode) {
+  const paddedSeasonNum = ("0" + episode.season).slice(-2);
+  const paddedEpisodeNum = ("0" + episode.number).slice(-2);
+  return `${episode.name} - S${paddedSeasonNum}E${paddedEpisodeNum}`;
 }
 
 function makePageForEpisodes(episodeList) {
-  const rootElem = document.getElementById("root");
-  rootElem.textContent = `Got ${episodeList.length} episode(s)`;
+  // make the episode dropdown list visible and hide the show dropdown list
+  toggleVisibility(episodeSelectHtml, showSelectHtml);
```

Comment:
"toggle" implies "swap" (i.e. that if you call toggle once it will change something, and call it again it will change it back) - I think what you're really doing in `toggleVisibility` is _hiding_ the elements - can you think of a more clear name for the function?